src/charm.py

```diff
@@ -32,7 +32,7 @@ def __init__(self, *args):
             name="loki",
             pebble_layer=self.pebble_layer,
             endpoints={"cluster": "loki-cluster"},
-            readiness_check_endpoint=f"http://localhost:{LOKI_PORT}/ready",
+            readiness_check_endpoint=f"http://{socket.getfqdn()}:{LOKI_PORT}/ready",
```

Comment:
If you'll have TLS, you'll likely need to use `https` endpoint. See https://github.com/canonical/tempo-worker-k8s-operator/blob/main/src/charm.py#L98 for reference.